### PR TITLE
Remove more unused code

### PR DESCRIPTION
**Pull request by guillaumerose**
This also include the removal of the migration code (from ConfigVersion 0 to 3). This version never changed for many years. 
https://github.com/code-ready/machine/pull/53/commits/d8c7cbd0918c980ff6d0f32983fa4512ea759f2c

3570 lines are left in drivers and libmachine. This PR removed 2000 lines.

### CODE REVIEW

File: libmachine/persist/filestore.go [hunk 1]
```diff
@@ -100,14 +100,6 @@ func (s Filestore) Exists(name string) (bool, error) {
 	return false, err
 }
 
-func (s Filestore) loadConfig(name string) (*host.Host, error) {
-	data, err := ioutil.ReadFile(filepath.Join(s.GetMachinesDir(), name, "config.json"))
```
Comment (cfergeau) [inline libmachine/persist/filestore.go]: The `filestore.go` changes seem unrelated to the ` Declare the variable where it is used ` commit.

Comment (guillaumerose) [reply libmachine/persist/filestore.go]: done!

**Review by gbraad — APPROVED** [review]
